platform/security/handler/security-handler-basic/src/main/java/org/codice/ddf/security/handler/basic/AbstractBasicAuthenticationHandler.java

```diff
@@ -0,0 +1,159 @@
+
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        String path = httpRequest.getServletPath();
+        LOGGER.debug("Handling request for path {}", path);
+
+        LOGGER.debug("Doing authentication and authorization for path {}", path);
+
+        BaseAuthenticationToken token = extractAuthenticationInfo(httpRequest);
+
+        // we found credentials, attach to result and return with completed status
+        if (token != null) {
+            handlerResult.setToken(token);
+            handlerResult.setStatus(HandlerResult.Status.COMPLETED);
+            return handlerResult;
+        }
+
+        // we didn't find the credentials, see if we are to do anything or not
+        if (resolve) {
+            doAuthPrompt(realm, (HttpServletResponse) response);
+            handlerResult.setStatus(HandlerResult.Status.REDIRECTED);
+        }
+
+        return handlerResult;
+    }
+
+    @Override
+    public HandlerResult handleError(ServletRequest servletRequest, ServletResponse servletResponse,
+            FilterChain chain) throws ServletException {
+        String realm = (String) servletRequest.getAttribute(ContextPolicy.ACTIVE_REALM);
+        doAuthPrompt(realm, (HttpServletResponse) servletResponse);
+        HandlerResult result = new HandlerResult(HandlerResult.Status.REDIRECTED, null);
+        result.setSource(realm + "-" + SOURCE);
+        LOGGER.debug("In error handler for basic auth - prompted for auth credentials.");
+        return result;
+    }
+
+    /**
+     * Return a 401 response back to the web browser to prompt for basic auth.
+     *
+     * @param realm
+     * @param response
+     */
+    private void doAuthPrompt(String realm, HttpServletResponse response) {
+        try {
+            response.setHeader(HttpHeaders.WWW_AUTHENTICATE,
+                    AUTHENTICATION_SCHEME_BASIC + " realm=\"" + realm + "\"");
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            response.setContentLength(0);
+            response.flushBuffer();
+        } catch (IOException ioe) {
+            LOGGER.debug("Failed to send auth response: {}", ioe);
+        }
+    }
+
+    protected BaseAuthenticationToken extractAuthenticationInfo(HttpServletRequest request) {
+
+        String realm = (String) request.getAttribute(ContextPolicy.ACTIVE_REALM);
+        String authHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+        if (authHeader == null || authHeader.equals("")) {
+            return null;
+        }
+
+        return extractAuthInfo(authHeader, realm);
+    }
+
+    /**
+     * Extract the Authorization header and parse into a username/password token.
+     *
+     * @param authHeader the authHeader string from the HTTP request
+     * @return the initialize UPAuthenticationToken for this username, password, realm combination (or null)
+     */
+    protected BaseAuthenticationToken extractAuthInfo(String authHeader, String realm) {
```

Comment:
I'd like to see us get in the habit of enforcing our constraints by contract, in this case by annotating these params as `NotNull`.